File: String/3451-string-compression-iii/string-compression-iii.java
class Solution {
    public String compressedString(String word) {
        StringBuilder comp = new StringBuilder();
        int count = 1;
        char present = word.charAt(0);
        
        for (int i = 1; i < word.length(); i++) {
            if (present == word.charAt(i) && count < 9) {
                count++;
            } else {
                comp.append(count).append(present);
                present = word.charAt(i);
                count = 1;
            }
        }
        
        comp.append(count).append(present);
        
        return comp.toString();
    }
}

File: String/3451-string-compression-iii/string-compression-iii.cpp
class Solution {
public:
    string compressedString(string word) {
        string comp = "";
        int count = 1;
        char present = word[0];

        for(int i = 1; i < word.size(); i++){
            if(present == word[i] && count < 9) count++;
            else{
                comp += to_string(count) + present;
                present = word[i];
                count = 1;
            }
        }

        comp += to_string(count) + present;

        return comp;
    }
};